tests/corpus/parsing/test_cha_parser.py

```diff
@@ -0,0 +1,95 @@
+import os
+import tempfile
+import pytest
+import requests
+import sktalk
+from sktalk.corpus.parsing.cha import ChaParser
+from sktalk.corpus.parsing.parser import Parser
+
+
+class TestParser:
+    milliseconds_timestamp = [
+        ["0", "00:00:00.000"],
+        ["1706326", "00:28:26.326"],
+        ["222222", "00:03:42.222"],
+        ["None", None]
+    ]
+
+    @pytest.mark.parametrize("milliseconds, timestamp", milliseconds_timestamp)
+    def test_to_timestamp(self, milliseconds, timestamp):
+        assert Parser._to_timestamp(milliseconds) == timestamp
+
+        with pytest.raises(ValueError, match="exceeds 24h"):
+            Parser._to_timestamp("987654321")
+
+        with pytest.raises(ValueError, match="negative"):
+            Parser._to_timestamp("-1")
+
+
+class TestChaParser:
+    urls = [
+        "https://ca.talkbank.org/data-orig/GCSAusE/01.cha",
+        "https://ca.talkbank.org/data-orig/GCSAusE/02.cha",
+        "https://ca.talkbank.org/data-orig/GCSAusE/03.cha"
+    ]
+
+    @pytest.fixture(params=urls)
+    def download_file(self, request):
+        remote = request.param
+        response = requests.get(remote, timeout=5)
+        response.raise_for_status()
+
+        ext = os.path.splitext(remote)[1]
+
+        with tempfile.NamedTemporaryFile(delete=False, suffix=ext) as temp_file:
+            temp_file.write(response.content)
+            temp_file_path = temp_file.name
+
+        yield temp_file_path
+
+        os.remove(temp_file_path)
```

Comment:
Nit-picky style comment: in order to avoid handling the file removal here, I _think_ you could yield the `temp_file.name` within the context:

```suggestion
        with tempfile.NamedTemporaryFile(suffix=ext) as temp_file:
            temp_file.write(response.content)
            yield temp_file.name
```
But needs to be tested, and is maybe not worth any effort.

I definitely like the solution using a fixture, BTW!